tests/rtm/test_rtm_client_functional.py

```diff
@@ -1,26 +1,51 @@
-# Standard Imports
+import asyncio
 import collections
 import unittest
-from unittest import mock
 
-# ThirdParty Imports
-import asyncio
 from aiohttp import web, WSCloseCode
-import json
 
-# Internal Imports
 import slack
 import slack.errors as e
-from tests.helpers import fake_send_req_args, mock_rtm_response
+from tests.rtm.mock_web_api_server import setup_mock_web_api_server, cleanup_mock_web_api_server
 
 
-@mock.patch("slack.WebClient._send", new_callable=mock_rtm_response)
 class TestRTMClientFunctional(unittest.TestCase):
-    async def echo(self, ws, path):
-        async for message in ws:
-            await ws.send(
-                json.dumps({"type": "message", "message_sent": json.loads(message)})
-            )
+    def setUp(self):
+        setup_mock_web_api_server(self)
+
+        self.loop = asyncio.new_event_loop()
+        asyncio.set_event_loop(self.loop)
+        task = asyncio.ensure_future(self.mock_server(), loop=self.loop)
+        self.loop.run_until_complete(asyncio.wait_for(task, 0.1))
+
+        self.client = slack.RTMClient(
+            token="xoxb-valid",
+            base_url="http://localhost:8765",
+            auto_reconnect=False,
+            run_async=False,
+        )
+        self.client._web_client = slack.WebClient(
+            token="xoxb-valid",
+            base_url="http://localhost:8888",
+            run_async=False,
+        )
+
+    def tearDown(self):
+        self.loop.run_until_complete(self.site.stop())
+        cleanup_mock_web_api_server(self)
+        if self.client:
+            # self.client.stop()
+
+            # If you see the following errors with #stop() method calls,  call `RTMClient#async_stop()` instead
+            #
+            # /python3.8/asyncio/base_events.py:641:
```

Comment:
This is unrelated to the code review suggestions. The tests have been passed but I overlooked this warning for two test cases.